lib/app.dart

```diff
@@ -0,0 +1,97 @@
+import 'dart:html' as html;
+
+html.DivElement get content => html.querySelector("#content");
+html.DivElement get container => html.querySelector("#container");
+html.DivElement get interactive => html.querySelector("#interactive");
+html.DivElement get progress => html.querySelector("#progress");
+html.ButtonElement get prevButton => html.querySelector("#prev-button");
+html.ButtonElement get nextButton => html.querySelector("#next-button");
+
+class App {
+  int currentPage = 0;
+  int maxPageCount = 10;
+
+  App() {
+    _listenToPrevClick();
+    _listenToNextClick();
+    _listenToShortcuts();
+
+    _showContent(0);
+    _renderProgress();
+    container.classes.toggle("hidden", false);
+  }
+
+  void _listenToPrevClick() {
+    prevButton.onClick.listen((_) => _goToSlide(--currentPage));
+  }
+
+  void _listenToNextClick() {
+    nextButton.onClick.listen((_) => _goToSlide(++currentPage));
+  }
+
+  void _listenToShortcuts() {
+    html.window.onKeyUp.listen((evt) {
+      switch (evt.keyCode) {
+        case html.KeyCode.RIGHT:
+          _goToSlide(++currentPage);
+          break;
+        case html.KeyCode.LEFT:
+          _goToSlide(--currentPage);
+          break;
+        default:
+      }
+    });
+  }
+
+  void _goToSlide(slideNum) {
+    if (slideNum < 0 || slideNum > maxPageCount - 1) {
+      return;
+    }
+
+    currentPage = slideNum;
+    _showContent(currentPage);
+    _renderProgress();
+    _updateButtonsState();
+  }
+
+  void _updateButtonsState() {
+    if (currentPage <= 0) {
+      prevButton.disabled = true;
+    } else {
+      prevButton.disabled = false;
+    }
+
+    if (currentPage >= maxPageCount - 1) {
```

Comment:
Doing something a comparison with a -1 is a bit confusing to read and reminds me of the nastiness of off-by-one errors. May I suggest creating a getter for a `firstPageIndex` that just returns 0 and a `lastPageIndex` that does the `maxPageCount - 1` for you? If you do that, feel free to ignore my previous comment in order to keep the -1.

Reply:
Ah first and last pageIndex is much cleaner. Have changed it, will make a note for the future as well!